File: 0219/1259.cpp
#include <iostream>
using namespace std;
int get_v[40][2];
int ans[80];
int n;
void sol(int start, int end) {
	if (end == 2 * n - 1)
		return;
	for (int i = 1; i < n; i++) {
		if (get_v[i][0] != 0) {
			if (get_v[i][0] == ans[end]) {
				ans[end + 1] = get_v[i][0];
				ans[end + 2] = get_v[i][1];
				get_v[i][0] = 0;
				sol(start, end + 2);
				return;
			}
			if (get_v[i][1] == ans[start]) {
				for (int j = end; j >= 0; j--) {
					ans[j + 2] = ans[j];
				}
				ans[start] = get_v[i][0];
				ans[start + 1] = get_v[i][1];
				get_v[i][0] = 0;
				sol(start, end + 2);
				return;
			}
		}
	}
}
int main(void) {
	int T;
	cin >> T;
	for (int test_case = 1; test_case <= T; test_case++) {
		cin >> n;
		for (int i = 0; i < n; i++) {
			cin >> get_v[i][0] >> get_v[i][1];
		}
		ans[0] = get_v[0][0];
		get_v[0][0] = NULL;
		ans[1] = get_v[0][1];
		sol(0, 1);
		cout << '#' << test_case;
		for (int i = 0; i < 2 * n; i++) {
			cout <<' '<< ans[i];
		}
		cout << '\n';
	}
	return 0;
}

File: 0219/1259.py
import sys
sys.stdin = open('1259.txt', 'r')
sys.stdout = open('1259_out.txt', 'w')
T = int(input())


def sol(data):
    n = len(data)
    if n == 1:
        return
    for i in range(1, n):
        if data[i][0] == data[0][-1]:
            data[0] = data[0] + data[i]
            data[:] = data[:i]+data[i + 1:]
            sol(data)
            break
        if data[i][-1] == data[0][0]:
            data[i] = data[i] + data[0]
            data[:] = data[1:]
            sol(data)
            break


for test_case in range(1, T + 1):
    n = int(input())
    get_data = list(map(int, input().split()))
    get_data = [get_data[2 * i:2 * i + 2]for i in range(n)]
    sol(get_data)
    print(f'#{test_case} ' + ' '.join([str(i) for i in get_data[0]]))
# T = int(input())
# for test_case in range(1, T + 1):
#     n = int(input())
#     get_data = list(map(int, input().split()))
#     ans = get_data[0:2]
#     get_data = get_data[2:]
#     n -= 1
#     k = 0
#     while n > 0:
#         if k == 0:
#             for i in range(n):
#                 if get_data[2 * i + k] == ans[-1]:
#                     ans = ans + get_data[2 * i + k:2 * i + k + 2]
#                     get_data = get_data[:2 * i + k] + get_data[2 * i + k + 2:]
#                     n -= 1
#                     break
#             else:
#                 k = 1
#         elif k == 1:
#             for i in range(n):
#                 if get_data[2 * i + k] == ans[0]:
#                     ans = get_data[2 * i + k - 1:2 * i + k + 1] + ans
#                     get_data = get_data[:2 * i + k - 1] + get_data[2 * i + k + 1:]
#                     n -= 1
#                     break
#             else:
#                 n = 0
#     print(f'#{test_case} ' + ' '.join([str(i) for i in ans]))
